llvm/include/llvm/IR/IntrinsicsNVVM.td

```diff
@@ -964,6 +964,10 @@ let TargetPrefix = "nvvm" in {
       DefaultAttrsIntrinsic<[llvm_half_ty], [llvm_half_ty], [IntrNoMem]>;
   def int_nvvm_ex2_approx_f16x2 :
       DefaultAttrsIntrinsic<[llvm_v2f16_ty], [llvm_v2f16_ty], [IntrNoMem]>;
+  def int_nvvm_ex2_approx_bf16 :
+      DefaultAttrsIntrinsic<[llvm_bfloat_ty], [llvm_bfloat_ty], [IntrNoMem]>;
+  def int_nvvm_ex2_approx_bf16x2 :
+      DefaultAttrsIntrinsic<[llvm_v2bf16_ty], [llvm_v2bf16_ty], [IntrNoMem]>;
```

Comment:
Can we switch nvvm.ex2.approx to be an overloaded intrinsic? Better yet can we use the generic exp2 intrinsic with the `afn` fast-math-flag to represent this operation?

Reply:
I think for `log2` we should support `afn` to codegen to `lg2.approx`. But for `exp2`, `afn` wouldn't change anything. And the NVVM intrinsics seem to have convention for suffix names for types (`.f` -> `float`, `.f16x2` -> `<2 x half>`, etc) that I can't achieve by using overloading, so I extended them with the same convention.

Reply:
Are these intrinsics different from `llvm.exp2` than? If not, we should probably remove and auto-upgrade them, not add new ones.

Reply:
They're not different and I think they could be removed, _but_ we have internal code that relies on them.

I can avoid adding the `bf16` variants then so it'll encourage people to switch to using `llvm.exp2` going forward.

Reply:
Removed the `bf16` intrinsics.